columnflow/production/normalization.py

```diff
@@ -53,16 +134,32 @@ def normalization_weights_requires(self: Producer, reqs: dict) -> None:
     Adds the requirements needed by the underlying py:attr:`task` to access selection stats into
     *reqs*.
     """
-    # TODO: for actual sample stitching, we don't need the selection stats for that dataset, but
-    #       rather the one merged for either all datasets, or the "stitching group"
-    #       (i.e. all datasets that might contain any of the sub processes found in a dataset)
+    self.selection_stats_key = f"{'stitched_' if self.allow_stitching else 'norm_'}selection_stats"
```

Comment:
I don't quite get why there has to be a dynamic attribute that is used in the requirements dict. Maybe I'm missing something, but would it be possible to drop it?

Reply:
If you use the same key for both the stitched and unstitched reqs function, the requirements from one Producer will overwrite the requirements from the other ones, resulting in getting other requirements than expected for one of the two Producers.

Reply:
Very true. We should revisit the contextualized requirements/inputs again at some point...